pkg/metallb/bgppeer.go

```diff
@@ -31,10 +31,10 @@ type BGPPeerAdditionalOptions func(builder *BGPPeerBuilder) (*BGPPeerBuilder, er
 
 // NewBPGPeerBuilder creates a new instance of BGPPeer.
 func NewBPGPeerBuilder(
-	apiClient *clients.Settings, name, nsname, peerIP string, asn, remoteASN uint32) *BGPPeerBuilder {
+	apiClient *clients.Settings, name, nsname, peerIP string, localASN uint32, remoteASN interface{}) *BGPPeerBuilder {
```

Comment:
Maybe better to create additional func withDynamicAsn and remove remoteASN there. Because you don't really configure remoteASN here and you will not able to modify an existing bgppeer CR

Reply:
Created a WithDynamicASN
Please review